api/graphql/validating_list_field.py

```diff
@@ -0,0 +1,23 @@
+from django.core.exceptions import ValidationError
+from rest_framework import serializers
+
+
+class ValidatingListField(serializers.ListField):
+    """
+    Default ListField returns extremely unclear error if child items
+    do not contain the given value. This class runs a custom validation
+    before default implementation and provides better error messages
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def run_child_validation(self, data):
+        if self.child and self.child.choices:
+            allowed_values = list(self.child.get_choices().values())
+            for value in data:
+                if value not in allowed_values:
+                    raise ValidationError(
+                        f"{value} is not a valid value. Allowed values: {', '.join(allowed_values)}."
```

Comment:
Will this possibly print bunch of `pk`s that are existing and could be used? If so I think it might be a bit too much :thinking: 
We could check if the child is PrimaryKeyRelated field or some  other pk field and give different validation error message

Reply:
I didn't find any clever way to make this happen so I chose to follow the dumb way. Since the number of available values can be any number, I just check if it is less than five. It it is, we show the options, otherwise we don't.

What do you think? 😄

Reply:
Sounds reasonable :ok_hand: